src/Storages/StorageS3.cpp

```diff
@@ -186,7 +186,7 @@ class StorageS3Source::DisclosedGlobIterator::Impl : WithContext
 
     size_t getTotalSize() const
     {
-        return total_size;
+        return total_size.load(std:memory_order_relaxed);
```

Comment:
Alternatively, could've just locked mutex here, but AFAIS total_size is used only for progress approximation and not for any precise calculations.